clouddriver-google/src/main/groovy/com/netflix/spinnaker/clouddriver/google/deploy/ops/UpsertGoogleImageTagsAtomicOperation.groovy

```diff
@@ -0,0 +1,99 @@
+/*
+ * Copyright 2017 Google, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License")
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.netflix.spinnaker.clouddriver.google.deploy.ops
+
+import com.google.api.services.compute.model.*
+import com.netflix.spinnaker.clouddriver.data.task.Task
+import com.netflix.spinnaker.clouddriver.data.task.TaskRepository
+import com.netflix.spinnaker.clouddriver.google.config.GoogleConfigurationProperties
+import com.netflix.spinnaker.clouddriver.google.deploy.GCEUtil
+import com.netflix.spinnaker.clouddriver.google.deploy.description.UpsertGoogleImageTagsDescription
+import com.netflix.spinnaker.clouddriver.orchestration.AtomicOperation
+import org.springframework.beans.factory.annotation.Autowired
+
+/**
+ * Update the set of labels defined on an image. The newly-specified key-value pairs are added to the existing set.
+ * Values can be empty.
+ *
+ * Uses {@link https://cloud.google.com/compute/docs/reference/beta/images/setLabels}
+ */
+class UpsertGoogleImageTagsAtomicOperation implements AtomicOperation<Void> {
+  private static final String BASE_PHASE = "UPSERT_IMAGE_TAGS"
+
+  private static Task getTask() {
+    TaskRepository.threadLocalTask.get()
+  }
+
+  private final UpsertGoogleImageTagsDescription description
+
+  @Autowired
+  private GoogleConfigurationProperties googleConfigurationProperties
+
+  @Autowired
+  String clouddriverUserAgentApplicationName
+
+  UpsertGoogleImageTagsAtomicOperation(UpsertGoogleImageTagsDescription description) {
+    this.description = description
+  }
+
+  /**
+   * curl -X POST -H "Content-Type: application/json" -d '[ { "upsertImageTags": { "imageName": "my-image-123", "tags": { "some-key-1": "some-val-2" }, "credentials": "my-account-name" } } ]' localhost:7002/gce/ops
+   */
+  @Override
+  Void operate(List priorOutputs) {
+    task.updateStatus BASE_PHASE, "Initializing upsert of image tags for $description.imageName..."
+
+    def credentials = description.credentials
+    def compute = credentials.compute
+    def project = credentials.project
```

Comment:
You can (provided you have the proper permissions). The `project` above is just passed so it can be added to the list of projects to search for the requested image. The project containing the resolved image is the one used when calling `.setLabels()` down below.